### PR TITLE
feat: flip vpkpp bsppp dependency

### PR DESCRIPTION
**Pull request by ashifolfi**
it's a reasonable assumption that parsing a bsp would include wanting to access the pack lump within, as important info and data can be found within that file.

However it is NOT a reasonable assumption that parsing VPKs would include wanting to parse whole BSP files. Thus it makes more sense that bsppp should depend on vpkpp, with extending the PackFile class done inside bsppp. That way you can have access to vpk parsing without the need of the whole BSP library.

This PR flips the dependency so that bsppp requires vpkpp and adds the BSP packfile format itself, so that vpkpp can be used without bsppp being included.

## Testing done:
I made sure it still compiled both fine. Outside of that I don't have the means to test much right now.

### CODE REVIEW

File: include/bsppp/BSPPack.h [hunk 1]
```diff
@@ -2,10 +2,12 @@
...
 #include <array>
 
-#include <bsppp/bsppp.h>
+#include <vpkpp/format/ZIP.h>
 
-#include "ZIP.h"
+#include "bsppp.h"
 
+// this extends off of vpkpp I think it's fine to keep it under the vpkpp  - Eden
+// also BSP already exists under the bsppp namespace
```
Comment (craftablescience) [inline include/bsppp/BSPPack.h:10]: honestly i think if youre going to have bsppp depend on vpkpp it wouldnt be a bad idea to merge those two classes

Comment (ashifolfi) [reply include/bsppp/BSPPack.h:10]: oh yeah fair it might be worth looking into.

Comment (ashifolfi) [reply include/bsppp/BSPPack.h]: should be done?

File: include/bsppp/bsppp.h [hunk 2]
```diff
@@ -4,15 +4,18 @@
 #include <cstddef>
 #include <optional>
 #include <string>
+#include <string_view>
 #include <vector>
 
+#include <vpkpp/format/ZIP.h>
```
Comment (craftablescience) [inline include/bsppp/bsppp.h:10]: note to self: move this below sourcepp include

File: src/bsppp/_bsppp.cmake [hunk 3]
```diff
@@ -1,4 +1,6 @@
 add_pretty_parser(bsppp
+        # BSP contains a packfile
```
Comment (craftablescience) [inline src/bsppp/_bsppp.cmake:2]: comment is unnecessary

**Comment by craftablescience** [comment]
im going to do this manually myself, pr is old and difficult to rebase, and i already deleted the BSP pack loader to add back after tholps pr lol

**Pull request closed without merging**